# EDA Project

<div class="alert alert-block alert-info">The EDA project in this course has four main parts to it: <br>
    
1. Project Proposal
2. Phase 1
3. Phase 2
4. Report

This notebook will be used for Project Proposal, Phase 1, and Phase 2. You will have specific questions to answer within this notebook for Project Proposal and Phase 1. You will also continue using this notebook for Phase 2. However, guidance and expectations can be found on Canvas for that assignment. The report is completed outside of this notebook and details can be found on the EDA Report page on Canvas.</div>

<div class="alert alert-block alert-danger"><b><font size=4>Read this before proceeding:</font></b>
    
1. Review the list of data sets and sources of data to avoid before choosing your data. This list is provided in the instructions for the Project Proposal assignment in Canvas.<br><br>  

2. It is expected that when you are asked questions requiring typed explanations you are to use a <b><u>markdown cell</u></b> to type your answers neatly. <b><u><i>Do not provide typed answers to questions as extra comments within your code.</i></u></b> Only provide comments within your code as you normally would, i.e. as needed to explain or remind yourself what each part of the code is doing.</div>

# Project Proposal

<div class="alert alert-block alert-info">The intent of this assignment is for you to share your chosen data file(s) with your instructor and provide general information on your goals for the EDA project.</div>

<div class="alert alert-block alert-success"><b>Step 1 (2 pts)</b>: Give a brief <i><u>description</u></i> of the source(s) of your data and include a <i><u>direct link</u></i> to your data.</div>

The sources that I'll be using for my project proposal are as follows:
1. https://datasets.imdbws.com/
2. https://www.the-numbers.com/movie/budgets

The data includes 3 separate files:

**movie_budget**: includes production costs, domestic and worldwide gross (The Numbers)

**title_basics**: includes runtimes, title's type, genre, primary and original title (IMDB)

**title_akas**: includes title of film, region, language, format type

<div class="alert alert-block alert-success"><b>Step 2 (2 pts)</b>: Briefly explain why you chose this data.</div>

I'm interested in determining in which season do movies sell most as it is my hunch that colder seasons are the most lucrative for film studios.

<div class="alert alert-block alert-success"><b>Step 3 (1 pt)</b>: Provide a brief overview of your goals for this project.</div>

The goal of this project is to determine if movie sales are impacted by seasonal changes in the northern hemisphere particularly the US

<div class="alert alert-block alert-success"><b>Step 4 (1 pt)</b>: Read the data into this notebook.</div>

In [7]:
import pandas as pd

# Set up notebook to display multiple outputs in one cell
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

In [134]:
# read data

movie_budget = pd.read_csv('movie_budget.csv')
title_basics = pd.read_csv('title.basics.tsv', sep='\t')
title_akas = pd.read_csv('title.akas.tsv', sep='\t')

/Users/chelciedealmeida/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3165: DtypeWarning: Columns (7) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


<div class="alert alert-block alert-success"><b>Step 5 (1 pt)</b>: Inspect the data using the <b>info(&nbsp;)</b>, <b>head(&nbsp;)</b>, and <b>tail(&nbsp;)</b> methods.</div>

In [90]:
# TODO: Use the info() method to determine to inspect the variable (column) names, the number of non-null values,
#       and the data types for each variable.
movie_budget.info() 
title_basics.info() 
title_akas.info()

# TODO: Use the head() method to inspect the first five (or more) rows of the data
movie_budget.head() 
title_basics.head() 
title_akas.head()

# TODO: Use the tail() method to inspect the last five (or more) rows of the data
movie_budget.tail() 
title_basics.tail() 
title_akas.tail()


<class 'pandas.core.frame.DataFrame'>
Int64Index: 5281 entries, 0 to 5816
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   Release Date       5281 non-null   datetime64[ns]
 1   Movie              5281 non-null   object        
 2   Production Budget  5281 non-null   int64         
 3   Domestic Gross     5281 non-null   int64         
 4   Worldwide Gross    5281 non-null   int64         
dtypes: datetime64[ns](1), int64(3), object(1)
memory usage: 247.5+ KB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8648763 entries, 0 to 8648762
Data columns (total 9 columns):
 #   Column          Dtype 
---  ------          ----- 
 0   tconst          object
 1   titleType       object
 2   primaryTitle    object
 3   originalTitle   object
 4   isAdult         object
 5   startYear       object
 6   endYear         object
 7   runtimeMinutes  object
 8   genres          object
dtypes: object(9)


Release Date                                        Movie  \
0   2019-04-23                            Avengers: Endgame   
1   2011-05-20  Pirates of the Caribbean: On Stranger Tides   
2   2015-04-22                      Avengers: Age of Ultron   
3   2015-12-16         Star Wars Ep. VII: The Force Awakens   
4   2018-04-25                       Avengers: Infinity War   

   Production Budget  Domestic Gross  Worldwide Gross  
0          400000000       858373000       2797800564  
1          379000000       241071802       1045713802  
2          365000000       459005868       1395316979  
3          306000000       936662225       2064615817  
4          300000000       678815482       2048359754

tconst titleType            primaryTitle           originalTitle  \
0  tt0000001     short              Carmencita              Carmencita   
1  tt0000002     short  Le clown et ses chiens  Le clown et ses chiens   
2  tt0000003     short          Pauvre Pierrot          Pauvre Pierrot   
3  tt0000004     short             Un bon bock             Un bon bock   
4  tt0000005     short        Blacksmith Scene        Blacksmith Scene   

  isAdult startYear endYear runtimeMinutes                    genres  
0       0      1894      \N              1         Documentary,Short  
1       0      1892      \N              5           Animation,Short  
2       0      1892      \N              4  Animation,Comedy,Romance  
3       0      1892      \N             12           Animation,Short  
4       0      1893      \N              1              Comedy,Short

titleId  ordering                      title region language  \
0  tt0000001         1                 Карменсіта     UA       \N   
1  tt0000001         2                 Carmencita     DE       \N   
2  tt0000001         3  Carmencita - spanyol tánc     HU       \N   
3  tt0000001         4                 Καρμενσίτα     GR       \N   
4  tt0000001         5                 Карменсита     RU       \N   

         types     attributes isOriginalTitle  
0  imdbDisplay             \N               0  
1           \N  literal title               0  
2  imdbDisplay             \N               0  
3  imdbDisplay             \N               0  
4  imdbDisplay             \N               0

Release Date                          Movie  Production Budget  \
5809   2004-10-08                         Primer               7000   
5810   2006-05-26                         Cavite               7000   
5813   1999-04-02                      Following               6000   
5814   2005-07-13  Return to the Land of Wonders               5000   
5816   2005-08-05              My Date With Drew               1100   

      Domestic Gross  Worldwide Gross  
5809          424760           841926  
5810           70071            71644  
5813           48482           240495  
5814            1338             1338  
5816          181041           181041

tconst  titleType               primaryTitle  \
8648758  tt9916848  tvEpisode              Episode #3.17   
8648759  tt9916850  tvEpisode              Episode #3.19   
8648760  tt9916852  tvEpisode              Episode #3.20   
8648761  tt9916856      short                   The Wind   
8648762  tt9916880  tvEpisode  Horrid Henry Knows It All   

                     originalTitle isAdult startYear endYear runtimeMinutes  \
8648758              Episode #3.17       0      2010      \N             \N   
8648759              Episode #3.19       0      2010      \N             \N   
8648760              Episode #3.20       0      2010      \N             \N   
8648761                   The Wind       0      2015      \N             27   
8648762  Horrid Henry Knows It All       0      2014      \N             10   

                             genres  
8648758         Action,Drama,Family  
8648759         Action,Drama,Family  
8648760         Action,Drama,Family  
8648761                       Short  
8648762  Adventure,Animation,Comedy

titleId  ordering           title region language        types  \
30770399  tt9916852         5  Episódio #3.20     PT       pt           \N   
30770400  tt9916852         6  Episodio #3.20     IT       it           \N   
30770401  tt9916852         7    एपिसोड #3.20     IN       hi           \N   
30770402  tt9916856         1        The Wind     DE       \N  imdbDisplay   
30770403  tt9916856         2        The Wind     \N       \N     original   

         attributes isOriginalTitle  
30770399         \N               0  
30770400         \N               0  
30770401         \N               0  
30770402         \N               0  
30770403         \N               1

In [91]:
movie_budget.shape
title_basics.shape
title_akas.shape

(5281, 5)

(8648763, 9)

(30770404, 8)

<div class="alert alert-block alert-danger"><b>STOP HERE for your Project Proposal assignment. Submit your (1) original data file(s) along with (2) the completed notebook up to this point, and (3) the html file for grading and approval.</b></div>

<div class="alert alert-block alert-warning"><b>Instructor Feedback and Approval (3 pts)</b>: Your instructor will provide feedback in either the cell below this or via Canvas. You can expect one of the following point values for this portion.

<b>3 pts</b> - if your project goals and data set are both approved.<br>
<b>2 pts</b> - if your data set is approved but changes to your project goals (Step 3) are needed.<br>
<b>1 pt</b> - if your project goals are approved but your data set is not approved.<br>
<b>0 pts</b> - if neither your data set nor your project goals are approved.<br><br>
    
<i><u>As needed, follow your instructor's feeback and guidance to get on track for the remaining portions of the EDA project.</u></i>
</div>

# EDA Phase 1

<div class="alert alert-block alert-info">The overall goal of this assignment is to take all necessary steps to inspect the quality of your data and prepare the data according to your needs. For information and resources on the process of Exploratory Data Analysis (EDA), you should explore the <b><u>EDA Project Resources Module</u></b> in Canvas.

Once you’ve read through the information provided in that module and have a comfortable understanding of EDA using Python, complete steps 6 through 10 listed below to satisfy the requirements for your EDA Phase 1 assignment. **Remember to convert code cells provided to markdown cells for any typed responses to questions.**</div>

<div class="alert alert-block alert-success"><b>Step 6 (2 pts)</b>: Begin by elaborating in more detail from the previous assignment on why you chose this data?<br>
    
1. Explain what you hope to learn from this data. 
2. Do you have a hunch about what this data will reveal? (The answer to this question will be used in the Introduction section of your EDA report.)
</div>

1. I'd like to know if film studios are intentional about their releases with regard to the seasons people are most 
   are most inclined to go to movie theaters.
2. I have a hunch that the film industry in North America generates most of their revenue during the winter. 

<div class="alert alert-block alert-success"><b>Step 7 (2 pts)</b>: Discuss the popluation and the sample:<br>
    
1. What is the population being represented by the data you’ve chosen? 
2. What is the total sample size?
</div>

1. Film sales in the U.S
2. 6222 based on the movie budgets dataframe, title_basics sample size is 8648763 and title_akas sample size is          30770404 but data analysis is will be determined by films with availale budget and gross figures

<div class="alert alert-block alert-success"><b>Step 8 (2 pts)</b>: Describe how the data was collected. For example, is this a random sample? Are sampling weights used with the data?</div>

The data was collected from IMDB and The numbers. IMDB's list is extensive and therefore inclusive of every film with
accreditation on their platform. Data from The Numbers, however, is based on films with the "biggest" production budget, domestic, and worldwide gross. It is, however, important to note that "The Number" data set is reflective of data where movie figures are available and therefore may include some bias since the data provided isn't exhaustive of every film ever made and their respective financial data. The heterogeneity of the data suggests the sample is partially random therefore a stratified random sample.

<div class="alert alert-block alert-success"><b>Step 9 (4 pts)</b>: In the Project Proposal assignment you used the info(&nbsp;) method to inspect the variables, their data types, and the number of non-null values. Using that information as a guide, provide definitions of each of your variables and their corresponding data types, i.e. a data dictionary. Also indicate which variables will be used for your purposes.</div>

# **_Data Dictionary:_**


### NOTE: 
**_Most variables were imported as objects but will be converted to their respective data types as shown the data dictionary._**

#### Additional Notes:
##### * variables with Included = 'Y' symbol denote variables that will be used in this analysis
##### * variables with Included = 'N' symbol denote variables that will not be used in this analysis

#### movie_budget:

| Column Name       | Column Description                                        | Data Type | Included |
|-------------------|-----------------------------------------------------------|-----------|----------|
| Release Date      | date which the movie was released                         | date      | Y        |
| Movie             | Name of the movie                                         | str       | Y        |
| Production Budget | Cost to make the movie                                    | int       | Y        |
| Domestic Gross    | the gross or total amount the movie produced domestically | int       | Y        |
| Worldwide Gross   | the gross or total amount the movie produced worldwide    | int       | Y        |

* All the columns in the movie_budget df are crucial to analysis as they pertain to either date the film was released
  or the financials of the movies which are the primary variables used in this EDA

#### title_basics: As described by [IMDB](https://www.imdb.com/interfaces/) 

| Column Name    | Column Description                                                                                         | Data Type | Included |
|----------------|------------------------------------------------------------------------------------------------------------|-----------|----------|
| tconst         | alphanumeric unique identifier of the title                                                                | str       | Y        |
| titleType      | the type/format of the title                                                                               | str       | Y        |
| primaryTitle   | the more popular title / the title used by the filmmakers on promotional materials at the point of release | str       | Y        |
| originalTitle  | original title, in the original language                                                                   | str       | Y        |
| isAdult        | 0: non-adult title; 1: adult title                                                                         | int       | N        |
| startYear      | represents the release year of a title. In the case of TV Series, it is the series start year              | int       | N        |
| endYear        | TV Series end year. ‘\N’ for all other title types                                                         | int       | N        |
| runtimeMinutes | primary runtime of the title, in minutes                                                                   | int       | N        |
| genres         | includes up to three genres associated with the title                                                      | str       | N        |


* tconst is IMBDs unique identifier for movies
* titletype will be used to exclude any format that is not a movie
* Title fields will be used to compare titles with The Numbers data set.
* Start year is the release date of the film unless the format is != Movie then it signifies TV show, short etc.


#### title_akas: As described by [IMDB](https://www.imdb.com/interfaces/) 

| Column Name     | Column Description                                                                                                                                                                                                                              | Data Type | Included |
|-----------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------|----------|
| titleId         | a tconst, an alphanumeric unique identifier of the title                                                                                                                                                                                        | str       | Y        |
| ordering        | a number to uniquely identify rows for a given titleId                                                                                                                                                                                          | int       | N        |
| title           | the localized title                                                                                                                                                                                                                             | str       | Y        |
| region          | the region for this version of the title                                                                                                                                                                                                        | str       | Y        |
| language        | the language of the title                                                                                                                                                                                                                       | str       | N        |
| types           | Enumerated set of attributes for this <br>alternative title. One or more of the following: "alternative", "dvd", <br>"festival", "tv", "video", "working", "original", "imdbDisplay". New <br>values may be added in the future without warning | str       | N        |
| attributes      | Additional terms to describe this alternative title, not enumerated                                                                                                                                                                             | str       | N        |
| isOriginalTitle | 0: not original title; 1: original title                                                                                                                                                                                                        | int       | N        |

* title can also be used to compare movie titles with The Numbers dataset
* region will be used to eliminate records that are not within the parameters of this analysis



<div class="alert alert-block alert-success"><b>Step 10 (10 pts)</b>: For full credit in this problem you'll want to <i><u>take all necessary steps to report on the quality of the data</u></i> and <i><u>clean the data accordingly</u></i>. Some things to consider while doing this are listed below. <b>Depending on your data and goals, there may be additional steps needed than those listed here.</b>
    
1. Are there rows with missing or inconsistent values? If so, eliminate those rows from your data where appropriate.
2. Are there any outliers or duplicate rows? If so, eliminate those rows from your data where appropriate. 
At each stage of cleaning the data, state how many rows were eliminated.
3. Are you using all columns (variables) in the data? If not, are you eliminating those columns?
4. Consider some type of visual display such as a boxplot to determine any outliers. Do any outliers need removed? If so, how many were removed?

At each stage of cleaning the data, state how many rows were eliminated. <b><u><i>It is good practice to get the shape of the data before and after each step in cleaning the data and add typed explanations (in separate markdown cells) of the steps taken to clean the data.</i></u></b><br></div>
    
<div class="alert alert-block alert-info">Include the rest of your work below and insert cells where needed.</div>

### movie_budget

In [30]:
movie_budget.shape
movie_budget.value_counts()

(5817, 5)

Release Date  Movie                                       Production Budget  Domestic Gross  Worldwide Gross
1/1/1959      Solomon and Sheba                            $5,000,000         $11,000,000     $11,016,959       1
6/15/1990     Dick Tracy                                   $47,000,000        $103,738,726    $162,738,726      1
6/15/1977     A Bridge Too Far                             $26,000,000        $50,800,000     $50,800,000       1
6/15/1960     The Apartment                                $3,000,000         $18,600,000     $24,600,000       1
6/15/1905     The Nutcracker                               $19,000,000        $2,119,994      $2,119,994        1
                                                                                                               ..
12/3/2015     Under sandet                                 $6,000,000         $434,640        $4,959,039        1
              Speak No Evil                                $150,000           $0             

In [36]:
unkDates = movie_budget.loc[movie_budget['Release Date'] == 'Unknown']
unkDates

Release Date                            Movie Production Budget  \
247       Unknown                   Desert Warrior      $140,000,000   
811       Unknown                                b       $70,000,000   
1290      Unknown           Hannibal the Conqueror       $50,000,000   
1462      Unknown  Les Bronzés 3: amis pour la vie       $42,000,000   
1650      Unknown                 Down Under Cover       $40,000,000   
...           ...                              ...               ...   
5790      Unknown                         Flywheel           $20,000   
5796      Unknown             Stories of Our Lives           $15,000   
5799      Unknown                      Tin Can Man           $12,000   
5811      Unknown                  The Mongol King            $7,000   
5812      Unknown                           Red 11            $7,000   

     Domestic Gross Worldwide Gross  
247              $0              $0  
811              $0              $0  
1290             $0              $0  
1462             $0     $83,833,602  
1650             $0              $0  
...             ...             ...  
5790             $0              $0  
5796             $0              $0  
5799             $0              $0  
5811           $900            $900  
5812             $0              $0  

[90 rows x 5 columns]

In [37]:
# Remove rows with Release Date = Unknown. Should drop 90 rows
movie_budget = movie_budget.drop(unkDates.index)
movie_budget.shape

(5727, 5)

In [38]:
# change the release date column to date data types
movie_budget['Release Date'] = movie_budget['Release Date'].astype('datetime64[ns]') 

In [41]:
# Convert Production Budget to int and remove nondigit characters
movie_budget['Production Budget'] = movie_budget['Production Budget'].replace("[$,]", "", regex=True).astype(int)

In [44]:
# Convert Domestic Gross to int and remove nondigit characters
movie_budget['Domestic Gross'] = movie_budget['Domestic Gross'].replace("[$,]", "", regex=True).astype(int)

In [45]:
# Convert Worldwide Gross to int and remove nondigit characters
movie_budget['Worldwide Gross'] = movie_budget['Worldwide Gross'].replace("[$,]", "", regex=True).astype(int)

In [46]:
# Display df to assess the three financial fields
movie_budget.head()

Release Date                                        Movie  \
0   2019-04-23                            Avengers: Endgame   
1   2011-05-20  Pirates of the Caribbean: On Stranger Tides   
2   2015-04-22                      Avengers: Age of Ultron   
3   2015-12-16         Star Wars Ep. VII: The Force Awakens   
4   2018-04-25                       Avengers: Infinity War   

   Production Budget  Domestic Gross  Worldwide Gross  
0          400000000       858373000       2797800564  
1          379000000       241071802       1045713802  
2          365000000       459005868       1395316979  
3          306000000       936662225       2064615817  
4          300000000       678815482       2048359754

In [47]:
# Converting Movie object to string
movie_budget['Movie'] = movie_budget['Movie'].astype(str)

In [51]:
# Check to make sure the movie field doesn't contain blanks and NaNs
movie_budget[movie_budget['Movie'] == ''].index
pd.isna(movie_budget['Movie'])

Int64Index([], dtype='int64')

0       False
1       False
2       False
3       False
4       False
        ...  
5810    False
5813    False
5814    False
5815    False
5816    False
Name: Movie, Length: 5727, dtype: bool

In [54]:
# We can drop the Worldwide Gross column since we're focused in the US market
movie_budget.drop(['Worldwide Gross'], axis=1)

Release Date                                        Movie  \
0      2019-04-23                            Avengers: Endgame   
1      2011-05-20  Pirates of the Caribbean: On Stranger Tides   
2      2015-04-22                      Avengers: Age of Ultron   
3      2015-12-16         Star Wars Ep. VII: The Force Awakens   
4      2018-04-25                       Avengers: Infinity War   
...           ...                                          ...   
5810   2006-05-26                                       Cavite   
5813   1999-04-02                                    Following   
5814   2005-07-13                Return to the Land of Wonders   
5815   2015-09-29                         A Plague So Pleasant   
5816   2005-08-05                            My Date With Drew   

      Production Budget  Domestic Gross  
0             400000000       858373000  
1             379000000       241071802  
2             365000000       459005868  
3             306000000       936662225  
4             300000000       678815482  
...                 ...             ...  
5810               7000           70071  
5813               6000           48482  
5814               5000            1338  
5815               1400               0  
5816               1100          181041  

[5727 rows x 4 columns]

In [56]:
# Check that none of the movies have a production budget = 0 as they're most likely bad data
movie_budget.loc[movie_budget['Production Budget'] == 0]

# Check that none of the movies have a Domestic Gross = 0 as they're most likely bad data, 
# streaming movies or the movie is newly released and financial figures have yet to be released

movie_budget.loc[movie_budget['Domestic Gross'] == 0]

Empty DataFrame
Columns: [Release Date, Movie, Production Budget, Domestic Gross, Worldwide Gross]
Index: []

Release Date                     Movie  Production Budget  \
75     2020-09-04                     Mulan          200000000   
166    2019-11-01              The Irishman          159000000   
220    2021-11-04                Red Notice          150000000   
221    2022-02-03                  Moonfall          150000000   
246    2022-05-27         Top Gun: Maverick          140000000   
...           ...                       ...                ...   
5791   2013-01-04  All Superheroes Must Die              20000   
5792   2015-04-21             The Front Man              20000   
5793   2011-11-25                The Ridges              17300   
5806   2015-05-19          Family Motocross              10000   
5815   2015-09-29      A Plague So Pleasant               1400   

      Domestic Gross  Worldwide Gross  
75                 0         69965374  
166                0           910234  
220                0           173638  
221                0                0  
246                0                0  
...              ...              ...  
5791               0                0  
5792               0                0  
5793               0                0  
5806               0                0  
5815               0                0  

[446 rows x 5 columns]

In [57]:
# Drop movies with Domestic Gross = 0
movie_budget.shape
domesticDrop = movie_budget.loc[movie_budget['Domestic Gross'] == 0]
movie_budget = movie_budget.drop(domesticDrop.index)
movie_budget.shape

(5727, 5)

(5281, 5)

In [99]:
movie_budget.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5281 entries, 0 to 5816
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   Release Date       5281 non-null   datetime64[ns]
 1   Movie              5281 non-null   object        
 2   Production Budget  5281 non-null   int64         
 3   Domestic Gross     5281 non-null   int64         
 4   Worldwide Gross    5281 non-null   int64         
dtypes: datetime64[ns](1), int64(3), object(1)
memory usage: 247.5+ KB


### title_basics

In [92]:
title_basics.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8648763 entries, 0 to 8648762
Data columns (total 9 columns):
 #   Column          Dtype 
---  ------          ----- 
 0   tconst          object
 1   titleType       object
 2   primaryTitle    object
 3   originalTitle   object
 4   isAdult         object
 5   startYear       object
 6   endYear         object
 7   runtimeMinutes  object
 8   genres          object
dtypes: object(9)
memory usage: 593.9+ MB


In [93]:
# Drop three columns that we won't need
title_basics.drop(['isAdult', 'startYear', 'endYear'], axis=1, inplace=True)

# Getting shape to get dimension of df
title_basics.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8648763 entries, 0 to 8648762
Data columns (total 6 columns):
 #   Column          Dtype 
---  ------          ----- 
 0   tconst          object
 1   titleType       object
 2   primaryTitle    object
 3   originalTitle   object
 4   runtimeMinutes  object
 5   genres          object
dtypes: object(6)
memory usage: 395.9+ MB


In [94]:
title_basics.shape
title_basics.head()

(8648763, 6)

tconst titleType            primaryTitle           originalTitle  \
0  tt0000001     short              Carmencita              Carmencita   
1  tt0000002     short  Le clown et ses chiens  Le clown et ses chiens   
2  tt0000003     short          Pauvre Pierrot          Pauvre Pierrot   
3  tt0000004     short             Un bon bock             Un bon bock   
4  tt0000005     short        Blacksmith Scene        Blacksmith Scene   

  runtimeMinutes                    genres  
0              1         Documentary,Short  
1              5           Animation,Short  
2              4  Animation,Comedy,Romance  
3             12           Animation,Short  
4              1              Comedy,Short

In [95]:
# Check the titleType to determine what to exclude from the dataset
title_basics['titleType'].value_counts()

tvEpisode       6465119
short            853167
movie            600289
video            257070
tvSeries         220118
tvMovie          134681
tvMiniSeries      41798
tvSpecial         35779
videoGame         30295
tvShort           10445
tvPilot               2
Name: titleType, dtype: int64

In [96]:
# Check table count
title_basics.shape

# Drop rows where value != movie
title_basics = title_basics[title_basics['titleType'].isin(['movie']) == True]

#title_basics.drop(title_basics[title_basics.titleType.isin(["", ""])].index)

# Check table count
title_basics.info()

(8648763, 6)

<class 'pandas.core.frame.DataFrame'>
Int64Index: 600289 entries, 498 to 8648713
Data columns (total 6 columns):
 #   Column          Non-Null Count   Dtype 
---  ------          --------------   ----- 
 0   tconst          600289 non-null  object
 1   titleType       600289 non-null  object
 2   primaryTitle    600289 non-null  object
 3   originalTitle   600289 non-null  object
 4   runtimeMinutes  600289 non-null  object
 5   genres          600289 non-null  object
dtypes: object(6)
memory usage: 32.1+ MB


In [97]:
title_basics['titleType'].value_counts()

movie    600289
Name: titleType, dtype: int64

In [98]:
# converting titleType to Str
title_basics['titleType'] = title_basics['titleType'].astype(str)

# converting primaryTitle to Str
title_basics['primaryTitle'] = title_basics['primaryTitle'].astype(str)

# converting originalTitle to Str
title_basics['originalTitle'] = title_basics['originalTitle'].astype(str)

# converting genres to Str
title_basics['genres'] = title_basics['genres'].astype(str)

title_basics.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 600289 entries, 498 to 8648713
Data columns (total 6 columns):
 #   Column          Non-Null Count   Dtype 
---  ------          --------------   ----- 
 0   tconst          600289 non-null  object
 1   titleType       600289 non-null  object
 2   primaryTitle    600289 non-null  object
 3   originalTitle   600289 non-null  object
 4   runtimeMinutes  600289 non-null  object
 5   genres          600289 non-null  object
dtypes: object(6)
memory usage: 32.1+ MB


In [116]:
# Check runtimeMinutes
title_basics['runtimeMinutes'].value_counts()

N       222840
90       25596
80       11364
60       10678
85       10667
         ...  
761          1
418          1
2400         1
326          1
566          1
Name: runtimeMinutes, Length: 489, dtype: int64

In [119]:
title_basics.loc[title_basics['runtimeMinutes'] == 'N']

tconst titleType               primaryTitle  \
610      tt0000615     movie         Robbery Under Arms   
625      tt0000630     movie                     Hamlet   
668      tt0000675     movie                Don Quijote   
729      tt0000739     movie  El pastorcito de Torrente   
783      tt0000793     movie              Andreas Hofer   
...            ...       ...                        ...   
8648439  tt9916178     movie         Yesterday's Dreams   
8648443  tt9916188     movie                   Minotaur   
8648560  tt9916428     movie        The Secret of China   
8648652  tt9916620     movie          The Copeland Case   
8648692  tt9916706     movie            Dankyavar Danka   

                        originalTitle runtimeMinutes                 genres  
610                Robbery Under Arms              N                  Drama  
625                            Amleto              N                  Drama  
668                       Don Quijote              N                  Drama  
729         El pastorcito de Torrente              N                  Drama  
783                     Andreas Hofer              N                  Drama  
...                               ...            ...                    ...  
8648439            Yesterday's Dreams              N                      N  
8648443                      Minotaur              N               Thriller  
8648560  Hong xing zhao yao Zhong guo              N  Adventure,History,War  
8648652             The Copeland Case              N                  Drama  
8648692               Dankyavar Danka              N                 Comedy  

[222840 rows x 6 columns]

In [120]:
# Drop rows with runtime = N
dropRuntime = title_basics.loc[title_basics['runtimeMinutes'] == 'N']
title_basics.drop(dropRuntime.index)
title_basics.shape

tconst titleType  \
498      tt0000502     movie   
570      tt0000574     movie   
587      tt0000591     movie   
672      tt0000679     movie   
1172     tt0001184     movie   
...            ...       ...   
8648612  tt9916538     movie   
8648653  tt9916622     movie   
8648680  tt9916680     movie   
8648703  tt9916730     movie   
8648713  tt9916754     movie   

                                              primaryTitle  \
498                                               Bohemios   
570                            The Story of the Kelly Gang   
587                                       The Prodigal Son   
672                         The Fairylogue and Radio-Plays   
1172                               Don Juan de Serrallonga   
...                                                    ...   
8648612                                Kuambil Lagi Hatiku   
8648653        Rodolpho Teóphilo - O Legado de um Pioneiro   
8648680  De la ilusión al desconcierto: cine colombiano...   
8648703                                             6 Gunn   
8648713                     Chico Albuquerque - Revelações   

                                             originalTitle runtimeMinutes  \
498                                               Bohemios            100   
570                            The Story of the Kelly Gang             70   
587                                      L'enfant prodigue             90   
672                         The Fairylogue and Radio-Plays            120   
1172                               Don Juan de Serrallonga             58   
...                                                    ...            ...   
8648612                                Kuambil Lagi Hatiku            123   
8648653        Rodolpho Teóphilo - O Legado de um Pioneiro             57   
8648680  De la ilusión al desconcierto: cine colombiano...            100   
8648703                                             6 Gunn            116   
8648713                     Chico Albuquerque - Revelações             49   

                             genres  
498                               N  
570      Action,Adventure,Biography  
587                           Drama  
672               Adventure,Fantasy  
1172                Adventure,Drama  
...                             ...  
8648612                       Drama  
8648653                 Documentary  
8648680                 Documentary  
8648703                           N  
8648713                 Documentary  

[377449 rows x 6 columns]

(600289, 6)

In [128]:
title_basics.shape

# Drop Genres
title_basics.drop(['genres'], axis=1, inplace=True)

title_basics.shape

(600289, 6)

(600289, 5)

In [130]:
#Drop runtimeMinutes and duplicates
title_basics.drop(['runtimeMinutes'], axis=1, inplace=True)
title_basics.drop_duplicates

title_basics.info()

<bound method DataFrame.drop_duplicates of             tconst titleType  \
498      tt0000502     movie   
570      tt0000574     movie   
587      tt0000591     movie   
610      tt0000615     movie   
625      tt0000630     movie   
...            ...       ...   
8648653  tt9916622     movie   
8648680  tt9916680     movie   
8648692  tt9916706     movie   
8648703  tt9916730     movie   
8648713  tt9916754     movie   

                                              primaryTitle  \
498                                               Bohemios   
570                            The Story of the Kelly Gang   
587                                       The Prodigal Son   
610                                     Robbery Under Arms   
625                                                 Hamlet   
...                                                    ...   
8648653        Rodolpho Teóphilo - O Legado de um Pioneiro   
8648680  De la ilusión al desconcierto: cine colombiano...   
8648692       

<class 'pandas.core.frame.DataFrame'>
Int64Index: 600289 entries, 498 to 8648713
Data columns (total 4 columns):
 #   Column         Non-Null Count   Dtype 
---  ------         --------------   ----- 
 0   tconst         600289 non-null  object
 1   titleType      600289 non-null  object
 2   primaryTitle   600289 non-null  object
 3   originalTitle  600289 non-null  object
dtypes: object(4)
memory usage: 39.0+ MB


### title_akas

In [135]:
title_akas.head()

titleId  ordering                      title region language  \
0  tt0000001         1                 Карменсіта     UA       \N   
1  tt0000001         2                 Carmencita     DE       \N   
2  tt0000001         3  Carmencita - spanyol tánc     HU       \N   
3  tt0000001         4                 Καρμενσίτα     GR       \N   
4  tt0000001         5                 Карменсита     RU       \N   

         types     attributes isOriginalTitle  
0  imdbDisplay             \N               0  
1           \N  literal title               0  
2  imdbDisplay             \N               0  
3  imdbDisplay             \N               0  
4  imdbDisplay             \N               0

In [136]:
title_akas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30770404 entries, 0 to 30770403
Data columns (total 8 columns):
 #   Column           Dtype 
---  ------           ----- 
 0   titleId          object
 1   ordering         int64 
 2   title            object
 3   region           object
 4   language         object
 5   types            object
 6   attributes       object
 7   isOriginalTitle  object
dtypes: int64(1), object(7)
memory usage: 1.8+ GB


In [137]:
#Drop ordering, language, types, attributes, isOriginalTitle
title_akas.drop(['ordering','language','types','attributes','isOriginalTitle'], axis=1, inplace=True)
title_akas.info

<bound method DataFrame.info of             titleId                      title region
0         tt0000001                 Карменсіта     UA
1         tt0000001                 Carmencita     DE
2         tt0000001  Carmencita - spanyol tánc     HU
3         tt0000001                 Καρμενσίτα     GR
4         tt0000001                 Карменсита     RU
...             ...                        ...    ...
30770399  tt9916852             Episódio #3.20     PT
30770400  tt9916852             Episodio #3.20     IT
30770401  tt9916852               एपिसोड #3.20     IN
30770402  tt9916856                   The Wind     DE
30770403  tt9916856                   The Wind     \N

[30770404 rows x 3 columns]>

In [138]:
title_akas.head()

titleId                      title region
0  tt0000001                 Карменсіта     UA
1  tt0000001                 Carmencita     DE
2  tt0000001  Carmencita - spanyol tánc     HU
3  tt0000001                 Καρμενσίτα     GR
4  tt0000001                 Карменсита     RU

In [140]:
title_akas_df = title_akas.loc[title_akas['region'] == 'US']

In [141]:
title_akas_df.info()
title_akas_df.shape

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1295121 entries, 5 to 30770148
Data columns (total 3 columns):
 #   Column   Non-Null Count    Dtype 
---  ------   --------------    ----- 
 0   titleId  1295121 non-null  object
 1   title    1295121 non-null  object
 2   region   1295121 non-null  object
dtypes: object(3)
memory usage: 39.5+ MB


(1295121, 3)

In [144]:
title_akas_df.drop_duplicates(inplace=True)

<ipython-input-144-2cf4a02dc689>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  title_akas_df.drop_duplicates(inplace=True)


In [145]:
title_akas_df.shape

(1293783, 3)

In [147]:
title_akas_df.info()

tconst                          title region
5         tt0000001                     Carmencita     US
14        tt0000002         The Clown and His Dogs     US
33        tt0000005               Blacksmith Scene     US
36        tt0000005            Blacksmithing Scene     US
41        tt0000005            Blacksmith Scene #1     US
...             ...                            ...    ...
30770075  tt9916702  Loving London: The Playground     US
30770113  tt9916720                The Demonic Nun     US
30770115  tt9916720                      The Nun 2     US
30770132  tt9916756       Pretty Pretty Black Girl     US
30770148  tt9916764                             38     US

[1293783 rows x 3 columns]

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1293783 entries, 5 to 30770148
Data columns (total 3 columns):
 #   Column   Non-Null Count    Dtype 
---  ------   --------------    ----- 
 0   titleId  1293783 non-null  object
 1   title    1293783 non-null  object
 2   region   1293783 non-null  object
dtypes: object(3)
memory usage: 39.5+ MB


In [148]:
# merge title_basics and title_akas_df

title_df = pd.merge(title_basics, title_akas_df, left_on='tconst', right_on='titleId')

In [150]:
title_df.head()
title_df.info()

tconst titleType                    primaryTitle  \
0  tt0000574     movie     The Story of the Kelly Gang   
1  tt0000591     movie                The Prodigal Son   
2  tt0000630     movie                          Hamlet   
3  tt0000679     movie  The Fairylogue and Radio-Plays   
4  tt0000886     movie       Hamlet, Prince of Denmark   

                    originalTitle    titleId                           title  \
0     The Story of the Kelly Gang  tt0000574     The Story of the Kelly Gang   
1               L'enfant prodigue  tt0000591                The Prodigal Son   
2                          Amleto  tt0000630                          Hamlet   
3  The Fairylogue and Radio-Plays  tt0000679  The Fairylogue and Radio-Plays   
4                          Hamlet  tt0000886       Hamlet, Prince of Denmark   

  region  
0     US  
1     US  
2     US  
3     US  
4     US

<class 'pandas.core.frame.DataFrame'>
Int64Index: 304951 entries, 0 to 304950
Data columns (total 7 columns):
 #   Column         Non-Null Count   Dtype 
---  ------         --------------   ----- 
 0   tconst         304951 non-null  object
 1   titleType      304951 non-null  object
 2   primaryTitle   304951 non-null  object
 3   originalTitle  304951 non-null  object
 4   titleId        304951 non-null  object
 5   title          304951 non-null  object
 6   region         304951 non-null  object
dtypes: object(7)
memory usage: 18.6+ MB


In [151]:
# Write two dataframes to excel:

writer = pd.ExcelWriter('EDA_Phase1.xlsx', engine='xlsxwriter')

movie_budget.to_excel(writer, sheet_name='Sheet1')
title_df.to_excel(writer, sheet_name='Sheet2')

# Close the pandas excel writer
writer.save()

<div class="alert alert-block alert-danger"><b>STOP HERE for your EDA Phase 1 assignment. Submit your <i><u>cleaned</u></i> data file along with the completed notebook up to this point for grading.</b></div>

# EDA Phase 2

<div class="alert alert-block alert-info">All of your work for the EDA Phase 2 assignment will begin below here. Refer to the detailed instructions and expectations for this assignment in Canvas.</div>